### ISSUE TITLE
After upgrading to Laravel 5.4 User model references undefined method

### ISSUE THREAD

**Issue by hermanschutte** [ForestAdmin/forest-laravel#32]
I'm now getting the following error when trying to access the User model in the admin: 

```
BadMethodCallException in Builder.php line 2508:
Call to undefined method Illuminate\Database\Query\Builder::getForeignKey()
```

**Comment by arnaudbesnier** [comment]
Hi @hermanschutte We have a brand new Laravel 5.4 project with the automatically generated User model for new projects. I can easily retrieve my users in Forest. Can you provide a longer stack trace ? Also, did you change something in your codebase that could generate such an error?

Thanks

**Comment by hermanschutte** [comment]
Hi @arnaudbesnier,

I think you need to setup a Laravel 5.4 project that is a bit more complex than the default one. We have a one to one relationship to oAuthIdentity for each user that seems to be causing the issue. This is not something that was changed recently. It worked before upgrading to 5.4 but now Forest is not able to retrieve the User model anymore.

Here is the stack trace:

```BadMethodCallException in Builder.php line 2508:
Call to undefined method Illuminate\Database\Query\Builder::getForeignKey()
in Builder.php line 2508
at Builder->__call('getForeignKey', array()) in Builder.php line 1332
at Builder->__call('getForeignKey', array())
at call_user_func_array(array(object(Builder), 'getForeignKey'), array()) in Relation.php line 340
at Relation->__call('getForeignKey', array()) in ResourcesGetter.php line 58
at ResourcesGetter->addJoins(object(Builder)) in ResourcesGetter.php line 37
at ResourcesGetter->perform() in ResourcesController.php line 23
at ResourcesController->index(object(Request), 'user')
at call_user_func_array(array(object(ResourcesController), 'index'), array(object(Request), 'user')) in Controller.php line 55
at Controller->callAction('index', array(object(Request), 'user')) in ControllerDispatcher.php line 44
at ControllerDispatcher->dispatch(object(Route), object(ResourcesController), 'index') in Route.php line 203
at Route->runController() in Route.php line 160
at Route->run() in Router.php line 559
at Router->Illuminate\Routing\{closure}(object(Request)) in Pipeline.php line 30
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in AuthenticateForestUser.php line 28
at AuthenticateForestUser->handle(object(Request), object(Closure)) in Pipeline.php line 148
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in Pipeline.php line 53
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in HandleCors.php line 42
at HandleCors->handle(object(Request), object(Closure)) in Pipeline.php line 148
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in Pipeline.php line 53
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in Pipeline.php line 102
at Pipeline->then(object(Closure)) in Router.php line 561
at Router->runRouteWithinStack(object(Route), object(Request)) in Router.php line 520
at Router->dispatchToRoute(object(Request)) in Router.php line 498
at Router->dispatch(object(Request)) in Kernel.php line 174
at Kernel->Illuminate\Foundation\Http\{closure}(object(Request)) in Pipeline.php line 30
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in RobotsMiddleware.php line 14
at RobotsMiddleware->handle(object(Request), object(Closure)) in Pipeline.php line 148
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in Pipeline.php line 53
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in CheckForMaintenanceMode.php line 46
at CheckForMaintenanceMode->handle(object(Request), object(Closure)) in Pipeline.php line 148
at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in Pipeline.php line 53
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in Pipeline.php line 102
at Pipeline->then(object(Closure)) in Kernel.php line 149
at Kernel->sendRequestThroughRouter(object(Request)) in Kernel.php line 116
at Kernel->handle(object(Request)) in index.php line 53
at require('/xxx/public/index.php') in server.php line 133```

**Comment by arnaudbesnier** [comment]
@hermanschutte Thanks for the stack trace, I'll try to find some time to investigate next week.

**Comment by hermanschutte** [comment]
@arnaudbesnier this should be a priority for you. We paid for Forest and we need this resolved in order to be able to use it.

**Comment by arnaudbesnier** [comment]
Hi @hermanschutte, this is fix in the new 0.0.18 Forest version.
FYI, as you can see in the Pull Request, this was due to a breaking change in Eloquent Relations API in Laravel 5.4
